lib/oauth.js

```diff
@@ -532,3 +540,44 @@ internals.location = function (request, protocol, location) {
     var host = request.info.host || (info.host + ':' + info.port);
     return protocol + '://' + host + request.path;
 };
+
+
+// Provide own QS implementation for cross node version support
+
+internals.encodePrimitive = function (value) {
+
+    var type = typeof value;
+    if (type === 'boolean') {
```

Comment:
I've heard that V8 can better optimize `typeof var === 'value'` patterns, so it might be preferable do multiple `typeof`s instead of caching the type.